pkg/operator/ceph/reporting/reporting.go

```diff
@@ -51,7 +51,7 @@ func ReportReconcileResult(logger *capnslog.PackageLogger, recorder *k8sutil.Eve
 		logger.Errorf("failed to reconcile %s %q. %v", kind, nsName, err)
 
 		// 2. event
-		recorder.ReportIfNotPresent(obj, corev1.EventTypeWarning, string(cephv1.ReconcileFailed), err.Error())
+		recorder.Event(obj, corev1.EventTypeWarning, string(cephv1.ReconcileFailed), err.Error())
```

Comment:
if this is happening on every retry, maybe we should keep the `ReportIfNotPresent()` call?

Reply:
If there are multiple events, they will be collated together like so, with something similar to `(x4 over 8m46s)` indicating that they have happened multiple times and without adding loads of clutter:
```
  Warning  ReconcileFailed     8m11s               rook-ceph-object-controller  failed to create object store deployments: fail
ed to start rgw health checker for CephObjectStore "rook-ceph/my-store", will re-reconcile: failed to create bucket checker for
 CephObjectStore "rook-ceph/my-store": failed to create or retrieve rgw admin ops user: failed to create object user "rgw-admin
-ops-user". error code 1 for object store "my-store": failed to create s3 user. . : signal: interrupt
  Normal   ReconcileSucceeded  9s (x4 over 8m46s)  rook-ceph-object-controller  successfully configured CephObjectStore "rook-c
eph/my-store"
```

Reply:
ok .so this is happening regardless of whether we use `ReportIfNotPresent` or `Event`?
I guess we can just rely on this mechanism instead of adding another one.